src/mlpack/core/data/dataset_mapper.hpp

```diff
@@ -151,10 +151,10 @@ class DatasetMapper
    * Serialize the dataset information.
    */
   template<typename Archive>
-  void serialize(Archive& ar, const unsigned int /* version */)
+  void serialize(Archive& ar, std::uint32_t const /* version */)
```

Comment:
```suggestion
  void serialize(Archive& ar, const uint32_t /* version */)
```
Correct me if I'm wrong, but I believe that this should work throughout and matches the mlpack style a bit better.  It should be easy to catch in a regex too---I don't believe we use `uint32_t` anywhere else.

Reply:
Yes, of course, one regexp should fix all mlpack sources